ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_transaction_metering.sql

```diff
@@ -0,0 +1,20 @@
+--  Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+SELECT 'Add Transaction Metering: Creating table...';
+
+CREATE TABLE transaction_metering (
+    application_id VARCHAR NOT NULL,
+    action_count INTEGER NOT NULL,
+    from_timestamp BIGINT NOT NULL,
+    to_timestamp BIGINT NOT NULL,
+    from_ledger_offset VARCHAR NOT NULL,
```

Comment:
I am wondering if we  really need the `from_ledger_offset` and `from_timestamp` fields. A record is an aggregation over a number of individual transactions from an arbitrary time span. We cannot effectively use both the from_ and the to_fields, so why store them in the first place. I can see the value of the to_* fields in the crash recovery scenario, but same cannot be said about the from_* fields.

Reply:
We could get by with either of the `from` or `to` variants.  The time are there to help with the un-aggregated reporting.  the offset there to help with crash recovery and aggregation.   A combination of both offset and time will be required for aggregation.  Don't forget these records will in the end only stick around for about an hour.  I like the fact that if the transaction_metering table is viewed then there should be no offset overlap for the same application.

Reply:
Agree that for this PR there would be a single timestamp but two offsets.  If following the aggregation phase one of the offsets is not removed it will be removed as part of that PR.